xcdat/spatial_avg.py

```diff
@@ -324,83 +345,136 @@ def _get_weights(
             if domain_bounds is not None:
                 dom_bounds = domain_bounds.copy()
 
+                # validate domain bounds
+                self._validate_domain_bounds(dom_bounds)
+
                 if region_bounds is not None:
                     reg_bounds = np.array(region_bounds).astype(float)
                     if dim == "lon":
-                        # Forces longitude bounds linearity for elements where
-                        # the upper value is greater than the lower value (e.g.,
-                        # across prime meridian or dateline) to avoid issues
-                        # in operations/calculations that operate using the lon
-                        # bounds.
-                        dom_bounds, reg_bounds = self._force_lon_linearity(
-                            dom_bounds, reg_bounds
-                        )
-                        # If the region bounds are equal, it means all longitude
-                        # values are specified (e.g., (360, 360) == (0, 360)).
-                        # As a result, set the region lon bounds to the domain
-                        # lon bounds (which represents all values).
-                        reg_bounds = self._set_equal_lon_bounds_to_domain(
-                            domain_bounds, reg_bounds
+                        # Ensure that domain and regional bounds are both
+                        # in the range of 0 -> 360
+                        dom_bounds = self._swap_lon_axes(dom_bounds, to=360)
+                        reg_bounds = self._swap_lon_axes(reg_bounds, to=360)
+
+                        # Some grid cells cross the prime meridian (
+                        # e.g., [-1, 1]). Check for this special case. If it
+                        # applies, return the index of that grid cell
+                        # (otherwise None) and recreate the dims with two
+                        # grid cells ([0, 1] and [359, 360]).
+                        pmb_index, dom_bounds = self._align_longitude_to_360_axis(
+                            dom_bounds
                         )
+
+                        # If the region bounds are equal, it means all
+                        # longitude values are specified (e.g., (360, 360) ==
+                        # (0, 360)). As a result, set the region lon bounds to
+                        # [0, 360].
+                        if reg_bounds[1] - reg_bounds[0] == 0:
```

Comment:
This is replacing `_set_equal_lon_bounds_to_domain` since it is used once and is only a couple lines. But this causes the code coverage to go down - I'm not exactly sure how to test that this specific section of code runs as needed.

Reply:
@tomvothecoder - do you know how I can test this? This runs in `get_weights`, which returns weights (not `reg_bounds`) - so I can't directly see if the function is appropriately adjusting the domain longitude bounds here.

Reply:
I usually add VSCode breakpoints to the lines I want to debug to check variable values at runtime. This involves a debug script with a call to get_weights(), then clicking "Debug Line" for that call.

If your editor doesn't support this, you can install ipdb in your conda environment and import it within this file. Then you can add ipdb breakpoints to pause at that line, step forward, step into , etc.

Reply:
I am pretty confident this is working as I want it to (I could debug to verify)...but the issue is that the codecov report indicated that this isn't covered (I assume it wants a unit test that covers this portion of code?).

Reply:
This is the [codecov report](https://app.codecov.io/gh/XCDAT/xcdat/compare/152/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr%20comments&utm_term=XCDAT#D1L373).